pkg/cmd/join/exec.go

```diff
@@ -100,32 +101,37 @@ func (o *Options) run() error {
 		"join/service_account.yaml",
 	}
 
-	err = apply.ApplyDirectly(clientHolder, reader, o.values, "", files...)
+	out, err := apply.ApplyDirectly(clientHolder, reader, o.values, o.dryRun, "", files...)
 	if err != nil {
 		return err
 	}
+	output = append(output, out...)
 
-	err = apply.ApplyDeployment(kubeClient, reader, o.values, "", "join/operator.yaml")
+	out, err = apply.ApplyDeployment(kubeClient, reader, o.values, o.dryRun, "", "join/operator.yaml")
 	if err != nil {
 		return err
 	}
+	output = append(output, out...)
 
-	b := retry.DefaultBackoff
-	b.Duration = 100 * time.Millisecond
+	if !o.dryRun {
```

Comment:
a better way is you have

```
if o.dryRun {
  output things
  return
}

apply manifests
```

Reply:
no I can not return because I still want to capture the output for the next ApplyCustomResouces.
I just want to skip the WaitCRDToBeReady as the CRD will not be created because it is dryRun.